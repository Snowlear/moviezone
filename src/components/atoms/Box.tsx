import boxStyles from "./Box.module.css";

export default function Box({
  children,
  styles,
  keyValue,
  onMouseEnter = () => {},
  onMouseLeave = () => {},
  onClick = () => {},
}: BoxProps) {
  return (
    <div
      key={keyValue}
      className={boxStyles.box + (styles === undefined ? "" : " " + styles)}
      onMouseEnter={() => onMouseEnter()}
      onMouseLeave={() => onMouseLeave()}
      onClick={() => onClick()}
    >
      {children}
    </div>
  );
}

interface BoxProps {
  children: any;
  styles?: string;
  keyValue?: string;
  onMouseEnter?: Function;
  onMouseLeave?: Function;
  onClick?: Function;
}
