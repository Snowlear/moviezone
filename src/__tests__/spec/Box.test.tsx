import React from "react";
import renderer from 'react-test-renderer';
import Box from "../../components/atoms/Box";

describe("<Box />", () => {
  it("renders <Box /> component with data.", async () => {
    const tree = renderer
    .create(<Box>Test</Box>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });
});