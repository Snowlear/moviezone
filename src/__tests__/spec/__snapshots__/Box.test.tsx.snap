// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders <Box /> component with data. 1`] = `
<div
  className="undefined"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</div>
`;
